test/libsolidity/syntaxTests/inlineAssembly/name_already_taken.sol

```diff
@@ -0,0 +1,9 @@
+contract C {
+    function f() public pure {
+        assembly { let x := 1 let x := 2 }
+        assembly { function f() { function f() {} } }
+    }
```

Comment:
There are more related tests in **test/libsolidity/InlineAssembly.cpp**.

What is our strategy here? We could:
[1] copy/move more tests from **InlineAssembly.cpp** to syntax tests;
[2] add **InlineAssembly.cpp** to coverage scripts;
[3] add minimal number of tests in syntax tests, just to cover error codes somehow.

(It is a general question about Boost tests.)

Reply:
The InlineAssembly.cpp tests should be turned into yul tests.